WalletWasabi/WabiSabi/Backend/Banning/CoinVerifierAuditArchiver.cs

```diff
@@ -0,0 +1,148 @@
+using NBitcoin;
+using Nito.AsyncEx;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using WalletWasabi.Helpers;
+
+namespace WalletWasabi.WabiSabi.Backend.Banning;
+
+public class CoinVerifierAuditArchiver : IAsyncDisposable
+{
+	public CoinVerifierAuditArchiver(string directoryPath)
+	{
+		IoHelpers.EnsureDirectoryExists(directoryPath);
+		BaseDirectoryPath = directoryPath;
+	}
+
+	private string BaseDirectoryPath { get; }
+
+	private AsyncLock FileAsyncLock { get; } = new();
+
+	private object LogLinesLock { get; } = new();
+
+	private List<AuditLine> LogLines { get; } = new();
+
+	public void LogException(uint256 roundId, Exception exception)
+	{
+		var logArray = new string[]
+		{
+			$"{roundId}",
+			$"{exception.Message}"
+		};
+
+		AddLogLineAndFormatCsv(DateTimeOffset.UtcNow, AuditEventType.Exception, logArray);
+	}
+
+	public void LogRoundEvent(uint256 roundId, string message)
+	{
+		var logAsArray = new string[]
+		{
+			$"Round ID: {roundId}",
+			$"{message}"
+		};
+
+		AddLogLineAndFormatCsv(DateTimeOffset.UtcNow, AuditEventType.Round, logAsArray);
+	}
+
+	public void LogVerificationResult(CoinVerifyResult coinVerifyResult, Reason reason, ApiResponseItem? apiResponseItem = null, Exception? exception = null)
+	{
+		string details = "No details";
+
+		if (apiResponseItem is not null)
+		{
+			var reportId = apiResponseItem?.Report_info_section.Report_id;
+			var reportType = apiResponseItem?.Report_info_section.Report_type;
+			var ids = apiResponseItem?.Cscore_section.Cscore_info?.Select(x => x.Id) ?? Enumerable.Empty<int>();
+			var categories = apiResponseItem?.Cscore_section.Cscore_info.Select(x => x.Name) ?? Enumerable.Empty<string>();
+
+			var detailsArray = new string[]
+			{
+					reportId ?? "ReportID None",
+					reportType ?? "ReportType None",
+					ids.Any() ? string.Join(' ', ids) : "FlagIds None",
+					categories.Any() ? string.Join(' ', categories) : "Risk categories None"
+			};
+
+			var formattedLines = detailsArray.Select(line => line.Replace(':', '-'));
```

Comment:
Add some comments here.

Reply:
This line is totally redundant now. We don't separate anything `:` and neither the details have any `:`.

Reply:
Plus now the line below 

L70
```
details = string.Join(",", formattedLines);
```

do unnecessary work because `AddLogLineAndFormatCsv` will replace the `,` with ` ` (space).

Reply:
Fixed [471a829](https://github.com/adamPetho/WalletWasabi/pull/8/commits/471a82940c02af427e13a38f7c198a1365de8f01)